util/blobs/blobs.go

```diff
@@ -47,6 +79,20 @@ func DecodeBlobs(blobs []kzg4844.Blob) ([]byte, error) {
 		for fieldIndex := 0; fieldIndex < params.BlobTxFieldElementsPerBlob; fieldIndex++ {
 			rlpData = append(rlpData, blob[fieldIndex*32+1:(fieldIndex+1)*32]...)
 		}
+		var acc uint16
+		accBits := 0
+		for fieldIndex := 0; fieldIndex < params.BlobTxFieldElementsPerBlob; fieldIndex++ {
+			acc |= uint16(blob[fieldIndex*32]) << accBits
+			accBits += spareBlobBits
+			if accBits >= 8 {
+				rlpData = append(rlpData, uint8(acc))
+				acc >>= 8
+				accBits -= 8
+			}
+		}
+		if accBits != 0 {
+			return nil, fmt.Errorf("somehow ended up with %v spare accBits", accBits)
+		}
 	}
 	var outputData []byte
 	err := rlp.Decode(bytes.NewReader(rlpData), &outputData)
```

Comment:
we already have the 1000 byte buffer so I think it's fine, but do we need to take into account the RLP length prefix of the data in the max bytes encoded per blob?

Reply:
Yeah that's a good point, though I think we'll just fit in the 1000